urlfd6bd5edbe_AntonPoznyak_LogFileGenerator_tgz-pJ8-logfilegen_StartMePleaseJ8/src/logfilegen/forFile/WriterFile.java

```diff
@@ -0,0 +1,24 @@
+package logfilegen.forFile;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WriterFile {
+	private String file;
+	private String source;
+	public WriterFile(String file,String source){
+		this.file= file;
+		this.source = source;
+	}
+	
+	public void write(){
+		java.io.FileWriter fileWriter;
+		try{
+			fileWriter = new FileWriter(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fileWriter*. The resource is referenced at line 18. The resource is closed at line 19. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fileWriter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).